### ISSUE TITLE
 An import path cannot end with a '.d.ts' extension. Consider importing './typescript/t.js' instead.

### ISSUE THREAD

**Issue by Akhila-S96** [i18next/i18next#2048]
## 🐛 Bug Report

Using i18next on angular application. Followed the instructions present in the installation guide. Receiving the below errors at compilation of angular app

                          ~~~~~~~~~


Error: node_modules/i18next/index.d.ts:2:34 - error TS2691: An import path cannot end with a '.d.ts' extension. Consider importing './typescript/helpers.js' instead.

2 import type { $Dictionary } from './typescript/helpers.d.ts';
                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~

Error: node_modules/i18next/index.d.ts:14:8 - error TS2691: An import path cannot end with a '.d.ts' extension. Consider importing './typescript/options.js' instead.

14 } from './typescript/options.d.ts';
          ~~~~~~~~~~~~~~~~~~~~~~~~~~~


Error: node_modules/i18next/index.d.ts:16:43 - error TS2691: An import path cannot end with a '.d.ts' extension. Consider importing './typescript/t.js' instead.

16 import type { KeyPrefix, TFunction } from './typescript/t.d.ts';
                                             ~~~~~~~~~~~~~~~~~~~~~


Error: node_modules/i18next/index.d.ts:497:1 - error TS1383: Only named exports may use 'export type'.

497 export type * from './typescript/options.d.ts';
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Error: node_modules/i18next/index.d.ts:497:20 - error TS2691: An import path cannot end with a '.d.ts' extension. Consider importing './typescript/options.js' instead.

497 export type * from './typescript/options.d.ts';
                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~


Error: node_modules/i18next/index.d.ts:516:8 - error TS2691: An import path cannot end with a '.d.ts' extension. Consider importing './typescript/options.js' instead.

516 } from './typescript/options.d.ts';
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~

Error: node_modules/i18next/index.d.ts:517:1 - error TS1383: Only named exports may use 'export type'.

517 export type * from './typescript/t.d.ts';
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Error: node_modules/i18next/index.d.ts:517:20 - error TS2691: An import path cannot end with a '.d.ts' extension. Consider importing './typescript/t.js' instead.

517 export type * from './typescript/t.d.ts';
                       ~~~~~~~~~~~~~~~~~~~~~


Error: node_modules/i18next/index.d.ts:524:8 - error TS2691: An import path cannot end with a '.d.ts' extension. Consider importing './typescript/t.js' instead.

524 } from './typescript/t.d.ts';


## To Reproduce

Take your own Angular project or create a new one, i.e. with [the Angular cli](https://angular.io/guide/setup-local#install-the-angular-cli).

npx @angular/cli new my-app

Install some i18next dependencies:

[i18next](https://www.i18next.com/)
[angular-i18next](https://github.com/Romanchuk/angular-i18next/)
[i18next-browser-languagedetector](https://github.com/i18next/i18next-browser-languageDetector)
npm install i18next angular-i18next i18next-browser-languagedetector

Basically follow this link -> https://adrai.medium.com/unleash-the-full-power-of-angular-i18next-8571adc2282d until all set.

## Expected behavior

Application should compile successfully.

## Actual behavior
Receiving above error and app not compiling

## Your Environment

- *runtime version*: i.e. node v16.15.1, browser Google chrome
- *i18next version*: i.e. 23.5.1
- *os*: Windows
- *Angular - 14.0.0

**Comment by adrai** [comment]
https://github.com/i18next/i18next/pull/2012

**Comment by wzenker** [comment]
Has it been resolved yet? Why do I still encounter this issue even after installing version 23.6.0?

**Comment by adrai** [comment]
@pedrodurek can you confirm this change is necessary? (together with the mjs change)

**Comment by wzenker** [comment]
I'm not sure, but I don't know how to resolve this issue right now. The compilation is not passing.
![image](https://github.com/i18next/i18next/assets/21078368/bf6e2a08-3a34-435d-adca-97a0cd564115)

**Comment by wzenker** [comment]
I'm not sure, but I don't know how to resolve this issue right now. The compilation is not passing.

: An import path cannot end with a '.d.ts' extension. Consider importing './typescript/options.js' instead.

516 } from './typescript/options.d.ts'

**Comment by adrai** [comment]
how does your tsconfig look like?

**Comment by wzenker** [comment]
this is my config {
  "compilerOptions": {
    "target": "es5",
    "lib": ["esnext", "dom"],
    "jsx": "react",
    "importHelpers": true,
    "noEmitHelpers": true,
    "strict": true,
    "strictPropertyInitialization": false,
    "allowSyntheticDefaultImports": true,
    "moduleResolution": "node",
    "esModuleInterop": true,
    "inlineSourceMap": true,
    "experimentalDecorators": true,
    "removeComments": false,
    "noImplicitAny": true,
    "diagnostics": true,
    "baseUrl": "./",
    "paths": {
      "tslib": ["node_modules/tslib"]
    }
  },

  "include": ["src/**/*"],
  "exclude": ["node_modules"]
}

**Comment by adrai** [comment]
Which  TypeScript Version are you using? Did you try an older or newer one?

**Comment by wzenker** [comment]
I am currently using TypeScript version 4.5.2, and I have tried the latest version, but it still doesn't work.

**Comment by adrai** [comment]
please try with v23.7.0
